Program.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace task13
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int[][] nums = new int[10][];
+            int index = 0;
+            int lastDigit = 0;
+            nums[lastDigit] = new int[0];
+            Console.WriteLine("Входной файл:");
+            using (StreamReader reader = new StreamReader("input.txt"))
+            {

```

Comment:
В коде неверное имя входного файла:

> В строках файл _random_numbers.txt_ содержатся...

```suggestion
            using (StreamReader reader = new StreamReader("random_numbers.txt"))
            {

```

А лучше не задавать имя файла в коде, а запрашивать с консоли.